smart_open/s3.py

```diff
@@ -687,13 +666,13 @@ def to_boto3(self):
         `boto3.s3.Object` may not necessarily affect the current instance.
 
         """
+        if not resource:
```

Comment:
Safer & more idiomatic:

```suggestion
        if resource is None:
```